### DISCUSSION TITLE
how to test translation message on inscape

### DISCUSSION THREAD

**Discussion by abc8143** [inkstitch/inkstitch#2591] [Q&A]
Hello,  I'm changing the file messages_zh_CN.po and I want to test it on inscape, what should I do?, I have less knowledge about packing the code

**Comment by bkmgit** [comment]
Thanks for contributing. If you enable GitHub Actions in your fork, it
will produce a test release that you can install.

**Reply by abc8143** [reply]
Thank you so much!!! :D

**Comment by kaalleen** [comment]
Have a look over here https://inkstitch.org/developers/localize/

I'll invite you to become a contributor.

**Reply by abc8143** [reply]
wow, thank you for inviting me, I'm watching crowdin now  :D
In fact I'm not so confident, so I want to pack the code and test it first, hope there is a way that I can test
anyway, I will try my best :D yeah

**Reply by kaalleen** [reply]
You can create a branch off from main in our repo directly and push your update.
That will build a release which you will be able to test.

**Reply by kaalleen** [reply]
On the other hand there is nothing wrong with adding it to crowdin and correct it later. We are not even close to a next official release. So no worries.

**Reply by abc8143** [reply]
ok now there is a new branch name "abc8143/zh_cn"
Is that I change message.po and commit to branch, github action will pack the code, and after testing, at last I need to add all these changes to crowdin?

**Reply by abc8143** [reply]
I get a new installation from github action, it seems no word change, maybe it only accept change from crowdin (ο´･д･)
![Snipaste_2023-11-12_03-53-58](https://github.com/inkstitch/inkstitch/assets/49285033/ba449d8e-0692-4527-bd1a-958811b32b2d)
![Snipaste_2023-11-12_03-54-45](https://github.com/inkstitch/inkstitch/assets/49285033/6b02fd20-3a83-4bde-8bdd-05786603af38)

**Reply by kaalleen** [reply]
Oh, hm, I'm not exactly sure on this. @lexelby could you help us out?

**Reply by lexelby** [reply]
It took me a minute to figure this out!

@abc8143 Here's the version that was built from your branch: https://github.com/inkstitch/inkstitch/releases/tag/dev-build-abc8143-zh_cn

That one does have your new translation of "Spacing between rows".

If you want to translate on your computer faster, you can do a [manual installation](https://inkstitch.org/developers/inkstitch/manual-setup/) of Ink/Stitch for local development.  Then you can edit the files in `translations/`, **run `make locales`**, and then run Ink/Stitch to see your new translations.

**Reply by abc8143** [reply]
oh wow I can see change now thank you so much!!! :D

**Reply by kaalleen** [reply]
Oh good. I was a little bit confused why it wouldn't work without testing it.

If we can just simply merge the translations ... I'm not sure. I just merged a new translation string to main. Let's see what crowdin is going to do with it.

**Reply by abc8143** [reply]
emmm.....That is a bit embarrassed,,,(◎_◎;) 
at first I just ......emmmm..... tired of seeing so many english, check the words.... watch the preview diffredent.... , mark down , so I hurry want to change all the words into chinese, and then pack one install to my own inscape, then hurry run my new sewing machine
you can see that I created a fork, but,,emmm... yes I have less knowledge about python...git...don't know how to pack the code, afraid make something wrong
Some words like "bean stitch", I search before, after seeing some pictures and preview, and google translate, I think, what, this call bean stitch, I feel it like something call back stitch。。。（／_＼）If I just pack one to my inscape that cause nothing, now if I add this word without checking preview, and then so many people can see it, I can't imagine
so now I change the words and test, make sure the words correct first, after that commit
![微信图片_20231112192128](https://github.com/inkstitch/inkstitch/assets/49285033/f06aa798-8390-42da-afeb-05cdd76122ce)

**Reply by kaalleen** [reply]
Looks as if crowdin deletes translations that we simply merge into main.
But you can upload your .po file to crowdin as it seems, so you don't have to enter everything manually. Here is how that is supposed to work https://support.crowdin.com/uploading-translations/

**Reply by abc8143** [reply]
can this crowdin upload message_zh_cn.po, then automatically match id and string?

**Reply by kaalleen** [reply]
I would think so, haven't tried the process. The good thing is that you can't break anything. You can just try to upload the file to crowdin and see what it does with it ;)

**Reply by abc8143** [reply]
hmm finally I just copy and paste one by one

**Reply by abc8143** [reply]
Hi friend, do you know about this, I found that the installtion package from github action, translation only work on one word "edit", translation work correct when it is local code.
But use local code I can't open the dialog of "Simulator/Realistic Preview" ......     (╥_╥)

![Snipaste_2024-04-10_03-23-57](https://github.com/inkstitch/inkstitch/assets/49285033/63f89e80-4bd7-4519-8356-b166967f3859)

![image](https://github.com/inkstitch/inkstitch/assets/49285033/f16d9bb8-adf1-4cdd-993e-12bde1e2e103)
